src/cordova/test.action.mjs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import url from 'url';
+import os from 'os';
+import minimist from 'minimist';
+import path from 'path';
+import fs from 'fs/promises';
+import rmfr from 'rmfr';
+
+import {getIosDeploymentTarget} from '../build/get_ios_deployment_target.mjs';
+import {getRootDir} from '../build/get_root_dir.mjs';
+import {spawnStream} from '../build/spawn_stream.mjs';
+
+const APPLE_ROOT = path.join(getRootDir(), 'src', 'cordova', 'apple');
+const APPLE_LIBRARY_NAME = 'OutlineAppleLib';
+
+/**
+ * @description Tests the parameterized cordova binary (ios, macos).
+ *
+ * @param {string[]} parameters
+ */
+export async function main(...parameters) {
+  const {
+    _: [outlinePlatform],
+  } = minimist(parameters);
+
+  if (outlinePlatform !== 'macos' && outlinePlatform !== 'ios') {
+    throw new Error('Testing is only currently supported for Apple platforms.');
+  }
+
+  if (os.platform() !== 'darwin') {
+    throw new Error('Building an Apple binary requires xcodebuild and can only be done on MacOS');
+  }
+
+  const derivedDataPath = path.join(APPLE_ROOT, 'coverage');
+
+  await rmfr(derivedDataPath);
+  await spawnStream(
+    'xcodebuild',
+    'clean',
+    'test',
+    '-scheme',
+    APPLE_LIBRARY_NAME,
+    '-destination',
+    outlinePlatform === 'macos'
+      ? `platform=macOS,arch=${os.machine()}`
```

Comment:
Perhaps add the variant. I see it lists Mac Catalyst as a possible destination to match:
![image](https://user-images.githubusercontent.com/113565/222831823-8a8edd71-731a-4b17-bdb7-a5ecd5973fa1.png)

Man: 
![image](https://user-images.githubusercontent.com/113565/222831893-d21bfcc5-a53f-479a-a7b3-7e069864d10f.png)


```suggestion
      ? `platform=macOS,variant=macOS,arch=${os.machine()}`
```

Reply:
They all have the same id, so I guess it doesn't matter? Weird.
Just ignore my suggestion and omit the variant then...